src/Compilers/CSharp/Test/Semantic/Semantics/RefLocalsAndReturnsTests.cs

```diff
@@ -617,21 +617,21 @@ public class C
+                // (36,24): error CS8521: Cannot use a result of 'Test.Foo<char>(ref char)' in this context because it may expose variables referenced by parameter 'arg' outside of their declaration scope
                 //             return ref Foo(ref M1);
-                Diagnostic(ErrorCode.ERR_RefReturnCall, "Foo(ref M1)").WithArguments("Test.Foo<char>(ref char)", "arg").WithLocation(36, 24),
+                Diagnostic(ErrorCode.ERR_EscapeCall, "Foo(ref M1)").WithArguments("Test.Foo<char>(ref char)", "arg").WithLocation(36, 24),
                 // (41,32): error CS8169: Cannot return a member of local 'M2' by reference because it is not a ref local
                 //             return ref Foo(ref M2.x);
                 Diagnostic(ErrorCode.ERR_RefReturnLocal2, "M2").WithArguments("M2").WithLocation(41, 32),
-                // (41,24): error CS8164: Cannot return by reference a result of 'Test.Foo<char>(ref char)' because the argument passed to parameter 'arg' cannot be returned by reference
+                // (41,24): error CS8521: Cannot use a result of 'Test.Foo<char>(ref char)' in this context because it may expose variables referenced by parameter 'arg' outside of their declaration scope
                 //             return ref Foo(ref M2.x);
-                Diagnostic(ErrorCode.ERR_RefReturnCall, "Foo(ref M2.x)").WithArguments("Test.Foo<char>(ref char)", "arg").WithLocation(41, 24),
+                Diagnostic(ErrorCode.ERR_EscapeCall, "Foo(ref M2.x)").WithArguments("Test.Foo<char>(ref char)", "arg").WithLocation(41, 24),
                 // (46,32): error CS8168: Cannot return local 'M2' by reference because it is not a ref local
                 //             return ref Foo(ref M2).x;
                 Diagnostic(ErrorCode.ERR_RefReturnLocal, "M2").WithArguments("M2").WithLocation(46, 32),
-                // (46,24): error CS8165: Cannot return by reference a member of result of 'Test.Foo<Test.S1>(ref Test.S1)' because the argument passed to parameter 'arg' cannot be returned by reference
+                // (46,24): error CS8522: Cannot use a member of result of 'Test.Foo<Test.S1>(ref Test.S1)' in this context because it may expose variables referenced by parameter 'arg' outside of their declaration scope
                 //             return ref Foo(ref M2).x;
-                Diagnostic(ErrorCode.ERR_RefReturnCall2, "Foo(ref M2)").WithArguments("Test.Foo<Test.S1>(ref Test.S1)", "arg").WithLocation(46, 24),
+                Diagnostic(ErrorCode.ERR_EscapeCall2, "Foo(ref M2)").WithArguments("Test.Foo<Test.S1>(ref Test.S1)", "arg").WithLocation(46, 24),
                 // (58,24): error CS8170: Struct members cannot return 'this' or other instance members by reference
                 //             return ref this;
                 Diagnostic(ErrorCode.ERR_RefReturnStructThis, "this").WithArguments("this").WithLocation(58, 24)
```

Comment:
I don't understand this error because C is a class, not a struct.

Reply:
Strange. Will need to look into this. It should have returned a different error.

---
In reply to: [135401366](https://github.com/dotnet/roslyn/pull/21334#discussion_r135401366) [](ancestors = 135401366)

Reply:
Returning a different error now.

---
In reply to: [135663135](https://github.com/dotnet/roslyn/pull/21334#discussion_r135663135) [](ancestors = 135663135,135401366)

Reply:
I don't see the new error in iteration 27. Did you push it yet?

---
In reply to: [135894874](https://github.com/dotnet/roslyn/pull/21334#discussion_r135894874) [](ancestors = 135894874,135663135,135401366)